src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/activities/ActivitiesService.kt

```diff
@@ -36,22 +36,19 @@ class ActivitiesService(
   }
 
   fun createActivity(dto: CreateActivityRequest): CreateActivityResponse =
-
-    activityRepository.save(mapActivityModel(dto)).run {
-
-      payRates = mapRates(dto, this)
-
-      telemetryClient.trackEvent(
-        "activity-created",
-        mapOf(
-          "id" to courseActivityId.toString(),
-          "prisonId" to prison.id,
-        ),
-        null
-      )
-
-      CreateActivityResponse(courseActivityId)
-    }
+    mapActivityModel(dto)
+      .apply { payRates.addAll(mapRates(dto, this)) }
```

Comment:
an alternative approach that @Mjwillis often uses would be to add a "addPayRates" method on the parent that deals with setting "this" on each child entity. It would probably slightly improve readability and guess would go in "mapRates"

But TBH it is marginal so not sure